### PR TITLE
Make user store common for all the internal Apis

### PR DESCRIPTION
**Pull request by sdkottegoda**
## Purpose
Fixes https://github.com/wso2/micro-integrator/issues/1450

A user store can be configured by adding the following to the deployment.toml
```
[[internal_apis.users]]
user.name = "user1"
password = "pwd1"
```
The user store can be disabled by adding the following parameter to to the deplyoment.toml
```
[internal_apis.file_user_store]
enable = false
```

### CODE REVIEW

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java [hunk 1]
```diff
@@ -175,6 +185,48 @@ public static void loadInternalApis(String apiFilePath) {
...
+        } else {
+            userMap = null;
+        }
+    }
+
+    /**
+     * Populates individual users.
+     *
+     * @param users the parent element of users
+     * @return map of users against credentioals
```
Comment (Nirothipan) [inline components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java:204]:
```suggestion
     * @return map of users against credentials
```

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/test/resources/internal/http/api/internal-apis-without-user-store.xml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<!--
+  ~ Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Nirothipan) [inline components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/test/resources/internal/http/api/internal-apis-without-user-store.xml:2]: Shall we update the year

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java (hunk 1 above)
Comment (sdkottegoda) [reply components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java]: done

File: components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/test/resources/internal/http/api/internal-apis-without-user-store.xml (hunk 2 above)
Comment (sdkottegoda) [reply components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/test/resources/internal/http/api/internal-apis-without-user-store.xml]: done

**Pull request merged**